src/framework/workspace/internal/workspaceactioncontroller.cpp

```diff
@@ -49,6 +61,44 @@ void WorkspaceActionController::openConfigureWorkspacesDialog()
     setCurrentWorkspaceName(selectedWorkspace);
 }
 
+void muse::workspace::WorkspaceActionController::createNewWorkspace()
+{
+    prepareCurrentWorkspaceForChange();
+
+    IWorkspacePtrList workspaces = manager()->workspaces();
+
+    QStringList workspaceNames;
+    for (const IWorkspacePtr& workspace: workspaces) {
+        workspaceNames << QString::fromStdString(workspace->name());
+    }
+
+    UriQuery uri("muse://workspace/create");
+    uri.addParam("sync", Val(true));
+    uri.addParam("workspaceNames", Val(workspaceNames.join(',')));
+
+    RetVal<Val> obj = interactive()->open(uri);
+    if (!obj.ret) {
+        return;
+    }
+
+    QVariantMap meta = obj.val.toQVariant().toMap();
+    QString name = meta.value("name").toString();
+    IF_ASSERT_FAILED(!name.isEmpty()) {
+        return;
+    }
+
+    IWorkspacePtr newWorkspace = manager()->cloneWorkspace(manager()->currentWorkspace(), name.toStdString());
+    if (!newWorkspace) {
+        return;
+    }
+
+    workspaces.emplace_back(newWorkspace);
+
+    manager()->setWorkspaces(workspaces);
+
+    setCurrentWorkspaceName(name.toStdString());
+}
```

Comment:
It looks like all this should be in the manager... the controller itself usually doesn't do anything, it only knows who to assign it to